code/modules/mob/living/brain/posibrain.dm

```diff
@@ -196,3 +244,41 @@ GLOBAL_VAR(posibrain_notify_cooldown)
 		icon_state = "[initial(icon_state)]-occupied"
 	else
 		icon_state = initial(icon_state)
+
+//This Proc triggers when the Z level changes. If the Posi enters the station level, add it to the Job list.
+//If it leaves, remove it.
+/obj/item/mmi/posibrain/onTransitZ(old_z, new_z)
```

Comment:
if `isstation(old_z) == isstation(new_z)`, we can just do early return doing nothing.

* 0, 0: same. no station to no station
* 1, 0: station to non-station. make it removed.
* 0, 1: non-station to station. make it added.
* 1, 1: multiz station to itself. we do nothing.

Reply:
I wasn't going to add this, then made the code slightly DRYer, so it ended up being necessary.

Originally didn't add it, because the 4 cases resulted in basically the same number of conditional statements.
0, 0: 2 conditionals executed, early return.
0, 1: 3 conditionals executed, normal return
1, 0: 4 conditionals executed, Normal return
1, 1: 2 conditionals executed, early return.

The original method was: 
0, 0: 3 conditionals executed
0, 1: 2 conditionals executed
1, 0: 3 conditionals executed
1, 1: 3 conditionals executed

Since adding that conditional basically added nothing except for covering the remaining logic conditionals which looks bloated and doesn't really need to be there, I didn't add it, since the function would normally just fall through, and that one set positions line is harmless.

Either way, a bit refactored, and code is now complete.